### PR TITLE
Feature/assets

### PR DESCRIPTION
**Pull request by hundredark**
Add mvm assets api

### CODE REVIEW

File: apps/explorer/priv/repo/migrations/20221117075000_tokens_add_asset_id_column.exs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+defmodule Explorer.Repo.Migrations.TokensAddAssetIdColumn do
+  use Ecto.Migration
+
+  def change do
+    alter table("tokens") do
+      add(:asset_id, :text)
```
Comment (cedricfung) [inline apps/explorer/priv/repo/migrations/20221117075000_tokens_add_asset_id_column.exs]: Is it better to use mixin_asset_id for this field?

File: apps/block_scout_web/lib/block_scout_web/views/api/rpc/token_view.ex [hunk 2]
```diff
@@ -24,7 +34,7 @@ defmodule BlockScoutWeb.API.RPC.TokenView do
       "totalSupply" => to_string(token.total_supply),
       "decimals" => to_string(token.decimals),
       "contractAddress" => to_string(token.contract_address_hash),
-      "cataloged" => token.cataloged
```
Comment (cedricfung) [inline apps/block_scout_web/lib/block_scout_web/views/api/rpc/token_view.ex:27]: Why is this removed? Not used in the view?

File: apps/block_scout_web/lib/block_scout_web/views/api/rpc/address_view.ex [hunk 3]
```diff
@@ -208,6 +212,7 @@ defmodule BlockScoutWeb.API.RPC.AddressView do
     %{
       "balance" => to_string(token.balance),
       "contractAddress" => to_string(token.contract_address_hash),
+      "assetId" => token.asset_id,
```
Comment (cedricfung) [inline apps/block_scout_web/lib/block_scout_web/views/api/rpc/address_view.ex]: So this may change together to mixinAssetId.

File: apps/explorer/lib/explorer/chain.ex [hunk 4]
```diff
@@ -4727,6 +4757,18 @@ defmodule Explorer.Chain do
     Repo.stream_reduce(query, initial, reducer)
   end
 
+  def stream_erc20_token(initial, reducer) when is_function(reducer, 2) do
```
Comment (cedricfung) [inline apps/explorer/lib/explorer/chain.ex]: Not used function?

File: apps/indexer/lib/indexer/fetcher/contract_log.ex [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+      @defaults
+      |> Keyword.merge(mergeable_init_options)
+      |> Keyword.put(:state, state)
+
+    Supervisor.child_spec({BufferedTask, [{__MODULE__, merged_init_opts}, gen_server_options]}, id: __MODULE__)
+  end
+
+  @impl BufferedTask
+  def init(initial_acc, reducer, _) do
+    contracts = ["0x3c84b6c98fbeb813e05a7a7813f0442883450b1f"]
```
Comment (cedricfung) [inline apps/indexer/lib/indexer/fetcher/contract_log.ex]: Use a constant or a config option for the address?

File: apps/block_scout_web/lib/block_scout_web/views/api/rpc/token_view.ex (hunk 2 above)
Comment (hundredark) [reply apps/block_scout_web/lib/block_scout_web/views/api/rpc/token_view.ex:27]: Yes, cataloged is used as a flag to update name, symbol, decimals and total supply in tokens table.

File: apps/explorer/priv/repo/migrations/20221117075000_tokens_add_asset_id_column.exs (hunk 1 above)
Comment (hundredark) [reply apps/explorer/priv/repo/migrations/20221117075000_tokens_add_asset_id_column.exs]: Change to mixin_asset_id

File: apps/explorer/lib/explorer/chain.ex (hunk 4 above)
Comment (hundredark) [reply apps/explorer/lib/explorer/chain.ex]: Removed

File: apps/indexer/lib/indexer/fetcher/contract_log.ex (hunk 5 above)
Comment (hundredark) [reply apps/indexer/lib/indexer/fetcher/contract_log.ex]: Use constants to save registry contract address and topic.

File: apps/block_scout_web/lib/block_scout_web/views/api/rpc/address_view.ex (hunk 3 above)
Comment (hundredark) [reply apps/block_scout_web/lib/block_scout_web/views/api/rpc/address_view.ex]: Change to mixinAssetId.
